### PR TITLE
skeleton for data studio checkpoints

### PR DESCRIPTION
**Pull request by jimmypoms**
Fixes https://github.com/seqeralabs/tower-backlog/issues/1215

### CODE REVIEW

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jason-seqera) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:180]: Is _Allowed S3 Buckets_ a UI element?

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jason-seqera) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:141]:
```suggestion
When a data studio starts, a new checkpoint is created. Every change to the root filesystem is recorded in the checkpoint. Changes to mounted datasets are not included in the checkpoint.
```

Something like this I think is more clear, if I understand correctly.

**Review by jason-seqera — COMMENTED** [review]
@jimmypoms, I've left a few minor comments. Thanks for this!

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jimmypoms) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:180]: It is and it is also referred to in other parts of the documentation see https://docs.seqera.io/platform/23.4.0/compute-envs/aws-batch :)

Comment (jason-seqera) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:180]:
```suggestion
By default, AWS Batch compute environments that are created with Batch Forge restrict access to S3 to the working directory only, unless additional **Allowed S3 Buckets** are specified.
```

Comment (jason-seqera) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:180]: @jimmypoms, looks like we've deprecated Fusion v1, though. Should we suggest people rely on this feature to have access to additional buckets? I'm not familiar with any of the background on this.

**Comment by jason-seqera** [comment]
@justinegeffen @llewellyn-sl 

I _think_ this is ready for a review; I also adjusted the headers as these were mostly H3s and as the page starts with an H1, I think these are supposed to be H2s, but there could be a reason for H3s that I'm not aware of. I moved some content to the top; we might have a preference for reference-like content after the procedural content, though, I'm not sure.

Eventually I think this might get broken up into an entire section of the docs rather than this single private preview page.

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:32]: "The checkpoint is persisted on the compute environment's working directory"

Should add that this is defined when creating a compute environment and link to the CE work dir docs.

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:34]:
```suggestion
When you restart, or start a new, data studio from a previously created checkpoint, changes such as installed software packages and configuration files are restored and available in the data studio session. Changes to mounted datasets are not included in the checkpoint.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:41]:
```suggestion
- **running**: When a data studio session is **running** and your user role is greater than **Viewer**, you can connect to it, copy the data studio URL, or stop it. In addition, the session can continue to process requests/run computations in the absence of an ongoing connection.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:43]:
```suggestion
- **stopped**: When a session is stopped, the associated compute resources are deallocated and any unsaved state is lost. If your user role is at least `Maintainer` you can start a session or delete the data studio when it's in this state.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:46]:
```suggestion
## Add a data studio

This functionality is available to users with the **Maintainer** role and above.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:49]:
```suggestion
2. Select a container image template from the three options provided (JupyterLab, RStudio Server, or Visual Studio Code).
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:99]:
```suggestion
## Start a data studio

This functionality is available to users with the **Maintainer** role and above.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:109]:
```suggestion
## Connect to a data studio

This functionality is available to all user roles excluding the **Viewer** role.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:121]:
```suggestion
## Collaborate in a data studio

This functionality is available to all user roles excluding the **Viewer** role.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:126]:
```suggestion
Collaborators will need valid workspace permissions to connect to the running data studio (all user roles excluding the **Viewer** role).
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:129]:
```suggestion
## Stop a data studio

This functionality is available to all user roles with the **Maintainer** role and above.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:139]:
```suggestion
When you restart a stopped session, a new session with the same configuration is created, using the most recent checkpoint.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:137]:
```suggestion
## Restart a stopped data studio

This functionality is available to all user roles with the **Maintainer** role and above.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:141]:
```suggestion
## Start a new data studio from a checkpoint

This functionality is available to all user roles with the **Maintainer** role and above.

You can start a new data studio from an existing stopped data studio. This will inherit the history of the parent data studio checkpoint state. From the list of **stopped** data studios in your workspace, select the three dots next to the status message for the data studio and select **Start as new**. Alternatively, from the data studio detail page select the **Checkpoints** tab and choose a non-running checkpoint to start a new data studio from by clicking the three dots in the `Actions` column and selecting **Start as new**.

## Delete a data studio

This functionality is available to all user roles with the **Maintainer** role and above.
```

**Review by robnewman — CHANGES_REQUESTED** [review]

**Comment by jason-seqera** [comment]
@robnewman, thanks!

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jason-seqera) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:30]: @robnewman, I go back and forth on that, but I can definitely avoid disrupting the overall structure at this time and revert to what we had and approved previously.

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jason-seqera) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:32]: @robnewman, the work directory docs for SP or for the specific CE provider, AWS, ect?

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jason-seqera) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:34]: @robnewman, this was that attempt at concrete examples.

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:32]: > This should include concrete examples, like installing packages or writing files.
Ah - I see the concrete examples below. You can ignore this change.

**Comment by jason-seqera** [comment]
@robnewman, I've changed the role names to align with: https://docs.seqera.io/platform/23.4.0/administration/overview

**Review by robnewman — APPROVED** [review]

**Comment by justinegeffen** [comment]
@jimmypoms, is this PR able to be pushed to prod once it's approved or is it linked to a release? @jason-seqera proposed using milestones to manage docs aligned with releases and if this does need to wait, we should probably use the `do not merge` label?

**Comment by jason-seqera** [comment]
@justinegeffen, this one is for summit.

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:151]:
```suggestion
When you start a newly added data studio for the first time, it automatically creates a *checkpoint*. A checkpoint persists changes that you make to the data studio root filesystem. The checkpoint is persisted on the compute environment's pipeline work directory.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:125]:
```suggestion
To stop a running session, select the three dots next to the status message and then select **Stop**. The status will change from **running** to **stopped**. When a data studio session is stopped, the compute resources it's using are deallocated. You can stop a session at any time.

Stopping a running session creates a new checkpoint.
```

**Review by robnewman — CHANGES_REQUESTED** [review]
Minor edits that I missed on the last review

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (justinegeffen) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:155]:
```suggestion
When you restart a data studio, or start a new data studio from a previously created checkpoint, changes such as installed software packages and configuration files are restored and made available in the data studio session. Changes made to mounted datasets are not included in a checkpoint.
```

**Review by justinegeffen — COMMENTED** [review]
This is looking good! A few comments/questions.

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jason-seqera) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:44]: @robnewman, @jimmypoms, what do you think? Thanks!

**Comment by justinegeffen** [comment]
@jimmypoms, @robnewman: this is scheduled to merge today. Can you please review and approve? Thanks!

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jimmypoms) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:44]: Are we using `dataset` as a general term to refer to data sets and cloud buckets from data explorer (aka data links)?

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jimmypoms) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:137]: I'm a bit confused here with what we mean by restart :sweat_smile:. Is this just starting a session or are we referring to start as new from the list?

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jimmypoms) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:153]: Clarification: A checkpoint is created every time a data studio session starts, no matter whether it is existing or newly-added, or for the first time :)

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:44]:
```suggestion
By default, Data Studios has read and write permissions to any mounted data. This means that data can be written from the data studio session back to the cloud storage mounted. To stop potential data loss, only one data studio session per workspace can mount a unique data path. When adding a new data studio, data paths already mounted to other running data studios are unavailable. If a new file is uploaded to the cloud storage bucket path while a session is running, the file may not be available to the data studio session immediately.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:137]: This is just starting a session, not `Start as new`

Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:137]: Simplified:
```suggestion
When you restart a stopped session, the session uses the most recent checkpoint.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:139]: We're missing a whole section on starting a new data studio from a checkpoint @jason-seqera 

CC: @jimmypoms

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jimmypoms) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:137]: In that case, is there a reason for this section at all considering that there is a section for [Start a data studio](https://github.com/seqeralabs/docs/pull/26/files#diff-fe51911e951cfe92d20f903be2226b5c217b4d737d5904b4ffa83e6933c6a078R85) and a section for [Start a new data studio from a checkpoint](https://github.com/seqeralabs/docs/pull/26/files#diff-fe51911e951cfe92d20f903be2226b5c217b4d737d5904b4ffa83e6933c6a078R139)

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jason-seqera) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:163]: @robnewman, I've added a section for this. Let me know what other salient points we want to surface about this functionality. Thanks!

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:103]: Immutable by definition means cannot be changed, so have one or the other, but not both (redundant):

```suggestion
The following fields are immutable:
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:109]: This is only true for data studio sessions in a stopped state

```suggestion
To add a new data studio from an existing **stopped** data studio, complete the following steps:
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:120]:
```suggestion

Additionally, you can add a new data studio from any existing data studio checkpoint (excluding the currently running checkpoint). From the data studio detail page, select the **Checkpoints** tab and in the **Actions** column, select **Start as new**. This is useful for tertiary analysis experimentation without impacting the state of the original data studio.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:113]:
```suggestion
1. This opens the **Add data studio** form. Optionally customize the following fields:
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:115]:
```suggestion
    - **Description**: By default, the parent data studio name(s).
```

**Review by robnewman — CHANGES_REQUESTED** [review]
Almost there!

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jason-seqera) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:103]: @robnewman, lol, true.

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jason-seqera) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:113]: @robnewman, I might rephrase this based on: https://developers.google.com/style/procedures#steps-with-results-or-justifications

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (jason-seqera) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:120]: @robnewman, per Google Developer Style, we're minimizing parenthesis where possible: https://developers.google.com/style/parentheses

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:103]: This no longer makes sense here. "Additionally," does not mean anything.

```suggestion
You can add a new data studio from any existing data studio checkpoint except the currently running checkpoint. From the data studio detail page, select the **Checkpoints** tab and in the **Actions** column, select **Start as new data studio**. This is useful for tertiary analysis experimentation without impacting the state of the original data studio.
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [inline platform_versioned_docs/version-23.4.0/data/data-studios.mdx:105]:
```suggestion
When adding a new data studio from an existing data studio or data studio checkpoint, the following fields are immutable:
```

File: platform_versioned_docs/version-23.4.0/data/data-studios.mdx
Comment (robnewman) [reply platform_versioned_docs/version-23.4.0/data/data-studios.mdx:122]: You can remove the parentheses, but I still think this text block makes more sense below the "Adding" from a stopped data studio.

**Review by robnewman — APPROVED** [review]
Minor layout changes, but otherwise approved.

**Pull request merged**
